airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/message/Batch.kt

```diff
@@ -45,6 +45,9 @@ import java.nio.file.Path
  * // etc...
  * ```
  */
+// Consider a sealed interface here, if you haven't already.
+// Making things sealed makes for so much nicer pattern matching
+// and compile-time checks. It's not always possible to seal things, of course.
```

Comment:
The intended usage pattern for the implementor is something like

`sealed class MyBatch: Batch`
`data class MyLocalFile: StagedLocalFile, ByBatch`
`data class MySpecialTable: MyBatch()`
etc

So we're providing

Reply:
OK got it, makes sense.

This will be apparent when there are destinations built using this CDK but in my opinion this is a good thing to comment about in the javadoc.

Reply:
If you expand the comment above you should see a detailed explanation with example usage. Let me know if you think that's insufficient.